source/lib/cdk-textract-stack.ts

```diff
@@ -170,35 +198,93 @@ export class CdkTextractStack extends cdk.Stack {
+      this.resourceName("ElasticSearchLogGroup"),
+      {
+        logGroupName: this.resourceName("ElasticSearchLogGroup")
+      }
+    );
+
+    // Elasticsearch
+    let elasticSearch;
+
     const esEncryptionKey = new kms.Key(this, "esEncryptionKey", {
       enableKeyRotation: true
     });
-    // Elasticsearch
-    const elasticSearch = new es.CfnDomain(
-      this,
-      this.resourceName("ElasticSearchCluster"),
-      {
-        elasticsearchVersion: "6.5",
-        elasticsearchClusterConfig: {
-          instanceType: "m5.large.elasticsearch"
-        },
-        ebsOptions: {
-          ebsEnabled: true,
-          volumeSize: 20,
-          volumeType: "gp2"
-        },
-        encryptionAtRestOptions: {
-          enabled: true,
-          kmsKeyId: esEncryptionKey.keyId
+
+    if (!props.isCICDDeploy) {
+      elasticSearch = new es.CfnDomain(
+        this,
+        this.resourceName("ElasticSearchCluster"),
+        {
+          elasticsearchVersion: "6.5",
+          elasticsearchClusterConfig: {
+            instanceType: "m5.large.elasticsearch"
+          },
+          ebsOptions: {
+            ebsEnabled: true,
+            volumeSize: 20,
+            volumeType: "gp2"
+          },
+          encryptionAtRestOptions: {
+            enabled: true,
+            kmsKeyId: esEncryptionKey.keyId
+          }
         }
-      }
-    );
+      );
+    } else {
+      elasticSearch = new es.CfnDomain(
+        this,
+        this.resourceName("ElasticSearchCluster"),
+        {
+          elasticsearchVersion: "6.5",
+          elasticsearchClusterConfig: {
+            instanceType: "m5.large.elasticsearch",
+            instanceCount: 2,
+            dedicatedMasterEnabled: true,
+            zoneAwarenessEnabled: true,
+            zoneAwarenessConfig: {
+              availabilityZoneCount: 2
+            }
+          },
+          ebsOptions: {
+            ebsEnabled: true,
+            volumeSize: 20,
+            volumeType: "gp2"
+          },
+          encryptionAtRestOptions: {
+            enabled: true,
+            kmsKeyId: esEncryptionKey.keyId
+          },
+          nodeToNodeEncryptionOptions: {
+            enabled: true
+          },
+          logPublishingOptions: {
+            INDEX_SLOW_LOGS: {
+              cloudWatchLogsLogGroupArn: esLogGroup.logGroupArn,
+              enabled: true
+            },
+            SEARCH_SLOW_LOGS: {
+              cloudWatchLogsLogGroupArn: esLogGroup.logGroupArn,
+              enabled: true
+            }
+          }
+        }
+      );
+    }
+
+    const jobCompletionTopicKey = new kms.Key(this, "jobCompletionTopicKey", {
+      enableKeyRotation: true
+    });
+
     // SNS Topic
     const jobCompletionTopic = new sns.Topic(
       this,
       this.resourceName("JobCompletion"),
       {
-        displayName: "Job completion topic"
+        displayName: "Job completion topic",
+        masterKey: jobCompletionTopicKey
       }
     );
 
```

Comment:
You need to grant the SNS encryption key usage permissions. Without the permissions Textract will not be able to publish to the SNS topic and all PDF processing would end up in a failed status. This is a pretty tricky case because since the processing is done on the Textract's side, we would never have any logs for the failed/error state